LibTest/async/MultiStreamController/addStream_A04_t01.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Future addStream(
+///   Stream<T> source, {
+///   bool? cancelOnError,
+/// })
+/// Receives events from source and puts them into this controller's stream.
+///
+/// Returns a future which completes when the source stream is done.
+///
+/// Events must not be added directly to this controller using `add`,
+/// `addError`, `close` or `addStream`, until the returned future is complete.
+///
+/// Data and error events are forwarded to this controller's stream. A done
+/// event on the source will end the addStream operation and complete the
+/// returned future.
+///
+/// If `cancelOnError` is `true`, only the first error on `source` is forwarded
+/// to the controller's stream, and the `addStream` ends after this. If
+/// `cancelOnError` is `false`, all errors are forwarded and only a done event
+/// will end the `addStream`. If `cancelOnError` is omitted or null, it defaults
+/// to `false`.
+///
+/// @description Checks that without done event on source the `addStream`
+/// operation is not ended and the returned future is not completed.
+/// @author sgrekhov22@gmail.com
+
+import "dart:async";
+import "../../../Utils/expect.dart";
+
+main() {
+  asyncStart();
+  var source = StreamController<int>();
+  source.add(1);
+  source.add(2);
+  source.add(3);
+  // No source.close() here.
+
+  var stream = Stream<int>.multi((controller) {
+    controller.addStream(source.stream).then((_) {
+      Expect.fail("Future unexpectedly completed");
+    });
+  });
+  listen(stream);
+}
+
+void listen(Stream<int> stream) {
+  int eventsCounter = 1;
+  stream.listen((v) {
+    Expect.equals(eventsCounter++, v);
+    if (v == 3) {
+      Future.delayed(Duration(milliseconds: 100), asyncEnd);
```

Comment:
I guess it's difficult to obtain an absolute guarantee that this test will never run on hardware which is tiny enough, and overloaded enough, such that the delay of 100 ms is too short?

However, I'd also assume that the test will actually succeed even in the case where `asyncEnd` is invoked too early. Right?

If so then it should be fine. Otherwise we would need to think about potential flaking.